### PR TITLE
Added support for docker-compose

### LINKED ISSUE #25: Add support for invoking docker compose using a specified yaml file
Currently all the info needed to build and run a container needs to be specified manually.
A feature to specify the environment needed to deploy as a yml file which can then be used to deploy containers via docker compose would be very useful.

### PR DESCRIPTION
**Pull request by the-c0d3br34k3r**
Closes #25.
Now we can spin up multi-container docker environments, given a valid
docker compose yaml file.

Since ShellUtility was needed, jIcing was added as a jar dependecy to
this project instead of a git submodule. This was done as, if we were
to add it as a git submodule, then the maven packaging would have to
be changed to "pom", in which case we won't be able to execute the
unit tests via maven.

As of now there is only support for docker compose up and down.
Support for remaining compose commands can be added as needed later.

### CODE REVIEW

File: src/test/resources/splunk_postgres_compose.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+services:
+  db:
+    image: postgres:latest
+    environment:
+      POSTGRES_PASSWORD: initialPassword12345!
+    ports:
+      - "5432:5432"
+  splunk:
```
[the remark this answers is not in the document]
Comment (the-c0d3br34k3r) [reply src/test/resources/splunk_postgres_compose.yml:8]: Fixed.

File: pom.xml [hunk 2]
```diff
@@ -40,47 +40,78 @@
...
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.7.0</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-params</artifactId>
-            <version>5.5.1</version>
+            <version>5.7.0</version>
```
Comment (abprash) [inline pom.xml:76]: Btw why are we going back in versions?

**Review by abprash — COMMENTED** [review]
Why are we going back in versions?

File: pom.xml (hunk 2 above)
Comment (the-c0d3br34k3r) [reply pom.xml:76]: What do you mean? It's been changed to 5.7.0 from 5.5.1

**Review by abprash — APPROVED** [review]

**Pull request merged**